developer-hub/src/polkadot_sdk/mod.rs

```diff
@@ -0,0 +1,126 @@
+//! communication between parachains, but is intended to be extensible to other use cases as well.
+//!
+//! ### Summary
+//!
+//! The following diagram summarizes how components of the Polkadot-SDK work together:
+#![doc = simple_mermaid::mermaid!("../../../docs/mermaid/polkadot_sdk.mmd")]
+//!
+//! 1. A Substrate-based chain is a blockchain composed of a "Runtime" and a "Client". As noted
+//!    above, the "Runtime" is the application logic of the blockchain, and the "Client" is
+//!    everything else. See [`reference_docs::wasm_meta_protocol`] for an in-depth explanation of
+//!    this. The former is built with [`frame`], and the latter is built with Substrate client
+//!    libraries.
+//! 2. Polkadot is itself a Substrate-based chain, composed of the the exact same two components.
+//!    The Polkadot client code is in [`polkadot`], and the Polkadot runtimes are controlled by the
+//!    [Polkadot Fellowship](https://github.com/polkadot-fellows).
+//! 3. A parachain is a "special" Substrate based chain, whereby both the client and the runtime
+//!    components have became "Polkadot-aware" using Cumulus.
+//!
+//! ## History
+//!
+//! Substrate, Polkadot and Cumulus used to each have their own repository, each of which is now
+//! archived. For historical context about how they merged into this mono-repo, see:
+//!
+//! - <https://polkadot-public.notion.site/Polkadot-SDK-FAQ-fbc4cecc2c46443fb37b9eeec2f0d85f>
+//! - <https://forum.polkadot.network/t/psa-parity-is-currently-working-on-merging-the-polkadot-stack-repositories-into-one-single-repository/2883>
+//!
+//! ## Notable Upstream Crates:
+//!
+//! - [`parity-scale-codec`](https://github.com/paritytech/parity-scale-codec)
+//! - [`parity-db`](https://github.com/paritytech/parity-db)
+//! - [`trie`](https://github.com/paritytech/trie)
+//! - [`parity-common`](https://github.com/paritytech/parity-common)
+//!
+//! [`substrate`]: crate::polkadot_sdk::substrate
+//! [`frame`]: crate::polkadot_sdk::frame_runtime
+//! [`cumulus`]: crate::polkadot_sdk::cumulus
+//! [`polkadot`]: crate::polkadot_sdk::polkadot
+//! [`xcm`]: crate::polkadot_sdk::xcm
+
+/// Lean about Cumulus, the framework that transforms [`substrate`]-based chains into
+/// [`polkadot`]-enabled parachains.
+pub mod cumulus;
+/// Learn about FRAME, the framework used to build Substrate runtimes.
+pub mod frame_runtime;
+/// Learn about Polkadot as a platform.
+pub mod polkadot;
+/// Learn about different ways through which smart contracts can be utilized on top of Substrate,
+/// and in the Polkadot ecosystem.
+pub mod smart_contracts;
+/// Learn about Substrate, the main blockchain framework used in the Polkadot ecosystem.
+pub mod substrate;
+/// Index of all the templates that can act as first scaffold for a new project.
+pub mod templates;
+/// Learn about XCM, the de-facto communication language between different consensus systems.
+pub mod xcm;
```

Comment:
with xcm being moved to an independent repo, should be bother documenting it here?

Reply:
My ask from Keith and Cisco has been to just provide a small amount of high level information about how XCM fits into the rest of Polkadot SDK (I already cover this to some extent in `polkadot` crate), maybe a simple code snippet to "attract", and then link to their home-base.